src/federation.cpp

```diff
@@ -467,4 +520,65 @@ void ThreadBlockSigner(NodeContext& node)
     }
 }
 
+void HandleBlockSignMessage(CNode& from, CDataStream& msg, std::chrono::microseconds time_received, const std::atomic<bool>& interrupt_msg_proc)
+{
+    unsigned char code;
+    msg >> code;
+
+    switch (code) {
+        case BlockSignMsgType::VER:
+        case BlockSignMsgType::VERACK:
+        {
+            LOCK(cs_block_signer);
+
+            CPubKey other;
+            msg >> other;
+            if (!other.IsFullyValid()) {
+                LogPrint(BCLog::FEDERATION, "Received blocksign::VER%s message with invalid/malformed public key.  Ignoring.\n", (code == BlockSignMsgType::VERACK) ? "ACK" : "");
```

Comment:
bf079d15a6b1581191235f23bc2b53e1131c4034

This is repeated often enough that it seems sensible to just create a `std::string` above and use that. I.e. somewhere above,
```C++
std::string msgstr = (code == BlockSignMsgType::VERACK ? "VERACK" : "VER");
```

Reply:
Done